lib/recommendation-server.js

```diff
@@ -42,6 +43,12 @@ RecommendationServer.prototype = {
 
     this.events.subscribe('urlbar-change', this.search);
   },
+  maxCharsForRecommendation: function() {
+    const name = 'maxCharsForSearchSuggestions';
+    return this.urlbarPrefs.getPrefType(name) ?
+           this.urlbarPrefs.getIntPref(name) :
+           20;
+  },
```

Comment:
This looks great, but I would only run this pref check once, inside the init() method, and set the result on the object. We call the search function a lot, and I'm not sure if hitting the pref system would slow things down.

Because this initialization happens for each new window, I think we're fine to take this shortcut in a prototype. If we wanted to be exactly correct, we would additionally set up a preferences observer, but let's skip that.